sources/packages/api/src/route-controllers/institution-user/institution-user.institutions.controller.ts

```diff
@@ -0,0 +1,277 @@
+import {
+  InstitutionUserAPIOutDTO,
+  CreateInstitutionUserAPIInDTO,
+  UpdateInstitutionUserAPIInDTO,
+  UserActiveStatusAPIInDTO,
+  InstitutionUserStatusAPIOutDTO,
+  InstitutionUserDetailAPIOutDTO,
+} from "./models/institution-user.dto";
+import {
+  ApiForbiddenResponse,
+  ApiNotFoundResponse,
+  ApiTags,
+  ApiUnprocessableEntityResponse,
+} from "@nestjs/swagger";
+import BaseController from "../BaseController";
+import { InstitutionUserControllerService } from "./institution-user.controller.service";
+import { ClientTypeBaseRoute } from "../../types";
+import { PaginatedResults, getUserFullName } from "../../utilities";
+import { BCeIDAccountTypeCodes } from "../../services/bceid/bceid.models";
+import { InstitutionUserPaginationOptionsAPIInDTO } from "../models/pagination.dto";
+
+/**
+ * Institution user controller for institutions client.
+ * Methods related to user management of the users that are associated with an
+ * institution and have more permissions/authorization needed.
+ */
+@AllowAuthorizedParty(AuthorizedParties.institution)
+@Controller("institution-user")
+@ApiTags(`${ClientTypeBaseRoute.Institution}-institution-user`)
+export class InstitutionUserInstitutionsController extends BaseController {
+  constructor(
+    private readonly institutionService: InstitutionService,
+    private readonly institutionUserControllerService: InstitutionUserControllerService,
+    private readonly userService: UserService,
+    private readonly bceidAccountService: BCeIDService,
+  ) {
+    super();
+  }
+
+  /**
+   * Get details of an institution user who is logged in.
+   * @returns user details.
+   */
+  @Get("my-details")
+  async getMyInstitutionDetails(
+    @UserToken() token: IInstitutionUserToken,
+  ): Promise<InstitutionUserDetailAPIOutDTO> {
+    const userDetails = await this.userService.getUser(token.userName);
+    const user = {
+      userName: userDetails.userName,
+      firstName: userDetails?.firstName,
+      lastName: userDetails.lastName,
+      isActive: userDetails.isActive,
+      userFullName: getUserFullName(userDetails),
+      isAdmin: token.authorizations.isAdmin(),
+      email: userDetails.email,
+    };
+    const authorizations = {
+      institutionId: token.authorizations.institutionId,
+      authorizations: token.authorizations.authorizations.map(
+        (authorization) => {
+          return {
+            locationId: authorization.locationId,
+            userType: authorization.userType,
+            userRole: authorization.userRole,
+          };
+        },
+      ),
+    };
+    return {
+      user: user,
+      authorizations: authorizations,
+    };
+  }
+
+  /**
+   * Get institution user summary.
+   * @param paginationOptions pagination options.
+   * provided the institution id will be retrieved from the token,
+   * when available.
+   * @returns all filtered institution users.
+   */
+  @Get()
+  async searchUsers(
```

Comment:
The method name says search users. I assume it is getInstitutionuserSummary(as per comment description)?

Reply:
And it does search the users also, right? I will change it.